wpengine/faustjs
# feat: Pagination component (BH-856) and getting-started example

- Adds a Next/WPGraphQL-based Pagination component to display next/previous links.
- Demonstrates its use in the getting-started `category.tsx` template.

<img width="1447" alt="Screenshot 2021-02-22 at 14 21 07" src="https://user-images.githubusercontent.com/647669/108714008-49ef8c00-7519-11eb-8ded-4f74c7a674d4.png">

## To test
1. In WordPress, make sure you have 21 or more posts in the uncategorized category. (For example, via [wp post generate](https://developer.wordpress.org/cli/commands/post/generate/).) This is so you can see posts across three pages.
2. `npm run bootstrap && npm run dev`
3. Visit http://localhost:3000/category/uncategorized.

You should see clickable next/previous page links.

## Notes
- Uses [cursor-based pagination as recommended by WPGraphQL](https://www.wpgraphql.com/2020/03/26/forward-and-backward-pagination-with-wpgraphql/). The downside is that sites migrating from WP will lose `/page/2` style pagination URLs, because WPGraphQL does not support/encourage this.
- I implemented this in a way that does not depend on client-side requests, so that users do not have to fill in `NEXT_PUBLIC_WORDPRESS_URL`, and so that paginated URLs such as `http://localhost:3000/category/uncategorized/after/YXJyYXljb25uZWN0aW9uOjU3MTY5` are more easily cacheable.
